File: server/chips/vulcan2.js
'use strict';

var VulcanType = require(__dirname + '/types/vulcan'),
    extend = require(__dirname + '/extend');

function Vulcan2Constructor(io, config, game, character) {
    this.properties.codes = ['D', 'F', 'L'];
    this.properties.description = '4-Shot to pierce one panel';
    this.properties.name = 'Vulcan 2';
    this.properties.short = 'Vulcan2';
    this.properties.bullets = 4;
    this.properties.power = 15;
    this.properties.stars = 2;
}

module.exports = function Vulcan2(io, config, game, character) {
    return extend(Vulcan2Constructor, VulcanType, io, config, game, character);
};

File: chips/vulcan2.py
# -*- coding: utf-8 -*-
# This file is part of MMBN Online
# MMBN Online is free software: you can redistribute it and/or modify
# it under the terms of the GNU General Public License as published by
# the Free Software Foundation, either version 3 of the License, or
# (at your option) any later version.
# MMBN Online is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
# GNU General Public License for more details.
# You should have received a copy of the GNU General Public License
# along with MMBN Online.  If not, see <http://www.gnu.org/licenses/>.

# Copyright (C) 2008-2010 Chris Santiago and Brandon Evans.
# http://mmbnonline.net/

"""A version of the Chip class."""

from chips.types.vulcan import Chip as Parent

class Chip(Parent):
    def properties(self):
        self.codes = ('D', 'F', 'L')
        self.description = '4-Shot to pierce one panel'
        self.name = 'Vulcan2'
        self.power = 15
        self.shots = 4
        self.stars = 2